### PR TITLE
Adding showHidden properties to file dialogs to allow for showing hidden files/folders.

### PR DESCRIPTION
**Pull request by wayne-chen**
- Fixing file dialog in QML to allow for displaying hidden files/folders.

https://highfidelity.manuscript.com/f/cases/14515/#

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9993/HighFidelity-Beta-PR13212-1107b085cb80f4ab89fe4603c623a91ea42f4e44.apk

**Comment by wayne-chen** [comment]
To test:

- Compile Interface project (if not already done)
- Launch Interface
- Log in with High Fidelity credentials in Menu->Login/Sign Up
- Go to Marketplace
- Click My Purchases
- Type in passphrase for account (if prompted)
- In the "Hmm, your keys are different" screen, click "LOCATE MY KEYS"
- Hidden files/folders should be displayed (i.e. %APPDATA% folder in Windows 10)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9993/HighFidelity-Beta-PR13212-1107b085cb80f4ab89fe4603c623a91ea42f4e44.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9993/HighFidelity-Beta-Interface-PR13212-1107b085cb80f4ab89fe4603c623a91ea42f4e44.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9993/HighFidelity-Beta-PR13212-1107b085cb80f4ab89fe4603c623a91ea42f4e44.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9993/HighFidelity-Beta-Interface-PR13212-1107b085cb80f4ab89fe4603c623a91ea42f4e44.exe

File: interface/resources/qml/dialogs/FileDialog.qml [hunk 1]
```diff
@@ -325,6 +325,7 @@ ModalWindow {
             showDirsFirst: true
             showDotAndDotDot: false
             showFiles: !root.selectDirectory
+	    showHidden: true
```
Comment (howard-stearns) [inline interface/resources/qml/dialogs/FileDialog.qml]: whitespace

**Comment by howard-stearns** [comment]
Looks like either a bad push, or GitHub has gone completely berserk. 1043 changed files?

**Comment by wayne-chen** [comment]
I totally messed up something with git, and it went bad.  I've fixed it at the final merge.

To answer your earlier questions -
1.  QML doesn't have a concrete understanding/flexibility with multiple inheritance. Also, the way the file dialog UIs for tablet, desktop, and Android are all different so the multiple files are required.

2.  The `alias` keyword allows for setting the showHidden property for the FolderListModel, which displays directories in the file dialog.  In the final commit (disregard the other ones), it has this control available.

**Comment by wayne-chen** [comment]
Going to recreate a pull request because of the mishap.

**Pull request closed without merging**